src/lib/components/FileUpload/FileUpload.tsx

```diff
@@ -1,23 +1,159 @@
+        {!maxFiles || files.length < maxFiles ? (
+          <div className="file-upload__wrapper">
+            <div
+              {...getRootProps()}
+              aria-labelledby={label ? labelId : undefined}
+              className="file-upload"
+              data-testid="file-upload"
+            >
+              <button className="file-upload__button" type="button">
+                Drag and drop files here or click to upload
+              </button>
+            </div>
+          </div>
+        ) : null}
+        {error ? (
+          <p className="p-form-validation__message">
+            <strong>Error: </strong>
+            {error}
+          </p>
+        ) : null}
+        <div className="file-upload__files-list">
+          {rejectedFiles &&
+            rejectedFiles.map((rejection) => (
+              <span className="is-error" key={rejection.file.name}>
+                <div className="file-upload__file is-rejected">
+                  {rejection.file.name}
+                  <Button
+                    appearance="base"
+                    className="file-upload__file-remove-button"
+                    onClick={() => removeRejectedFile(rejection)}
+                    type="button"
+                  >
+                    <Icon name="close">Remove file</Icon>
+                  </Button>
+                </div>
+                {rejection.errors.map((error) => (
+                  <p
+                    className="p-form-validation__message"
+                    key={`${rejection.file.name}-${error.code}`}
+                  >
+                    {error.message}
+                  </p>
+                ))}
+              </span>
+            ))}
+          {files &&
+            files.map((file) => (
+              <div className="file-upload__file" key={file.name}>
+                {file.name}
+                <Button
+                  appearance="base"
+                  className="file-upload__file-remove-button"
+                  onClick={() => removeFile(file)}
+                  type="button"
+                >
+                  <Icon name="close">Remove file</Icon>
+                </Button>
+              </div>
+            ))}
+        </div>
+      </div>
     </div>
   );
 };
+
+export const FileUploadContainer = ({
+  error,
+  help,
+  label,
+  maxFiles,
+  maxSize,
+}: Pick<FileUploadProps, "error" | "help" | "label" | "maxFiles" | "maxSize">) => {
+  const [files, setFiles] = useState<File[]>([]);
+  const [rejectedFiles, setRejectedFiles] = useState<FileRejection[]>([]);
+
+  const onFileUpload = useCallback(
+    (acceptedFiles: File[], fileRejections: FileRejection[]) => {
+      setFiles([...files, ...acceptedFiles]);
+      setRejectedFiles([...rejectedFiles, ...fileRejections]);
+    },
+    [files, rejectedFiles],
+  );
+
+  const removeFile = (file: File) => {
+    const newFiles = [...files];
+    newFiles.splice(newFiles.indexOf(file), 1);
+    setFiles(newFiles);
+  };
```

Comment:
You can access a previous state using a setState callback, and then use filter to make it more concise.
```suggestion
  const removeFile = (file: File) =>
    setFiles((prevFiles) => prevFiles.filter((f) => f !== file));
```